### PR TITLE
P4RT 2.1 - Script updated to add interface type to DUT

### PR DESCRIPTION
**Pull request by mitaliy**
Updated go script for P4rt 2.1

Added a function configureInterfaceType() which updates the interface type with the port ID on the DUT.

### CODE REVIEW

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go [hunk 1]
```diff
@@ -92,10 +93,21 @@ func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
...
+	i.Type = if_type
+	return i
+}
+
 // configurePortId configures p4rt port-id on the DUT.
 func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
 	portName := dut.Port(t, "port1").Name()
-	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portId)
+	curr_intf := &oc.Interface{Name: ygot.String(portName)}
```
Comment (Ankur19) [inline feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go:106]: Inline the `Type` value to `ethernetCsmacd` here without adding the additional functions.

Comment (mitaliy) [reply feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go]: I have updated the code.

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ var (
 	npDesc       = "New Primary"
 	gdpEtherType = *ygot.Uint32(0x6007)
 	portId       = *ygot.Uint32(10)
+	if_type      = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
```
Comment (Ankur19) [inline feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go:34]: remove the variable and use it directly while building the struct.

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go [hunk 3]
```diff
@@ -92,10 +93,16 @@ func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
...
+// configurePortId configures p4rt port-id and interface type on the DUT.
 func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
 	portName := dut.Port(t, "port1").Name()
-	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portId)
+	curr_intf := &oc.Interface{Name: ygot.String(portName),
+		Type: if_type}
+	gnmi.Replace(t, dut, d.Interface(portName).Config(), curr_intf)
+
+	gnmi.Replace(t, dut, d.Interface(portName).Id().Config(), portId)
```
Comment (Ankur19) [inline feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go:104]: this gnmi.Replace is not needed. Instead add `Id` as part of the `curIntf` struct.

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go [hunk 4]
```diff
@@ -92,10 +93,16 @@ func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
 	gnmi.Replace(t, dut, gnmi.OC().Component(p4rtNode).Config(), &c)
 }
 
-// configurePortId configures p4rt port-id on the DUT.
+// configurePortId configures p4rt port-id and interface type on the DUT.
 func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
 	portName := dut.Port(t, "port1").Name()
-	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portId)
+	curr_intf := &oc.Interface{Name: ygot.String(portName),
```
Comment (Ankur19) [inline feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go:100]: Use mixed caps: https://google.github.io/styleguide/go/guide#mixedcaps

**Review by Ankur19 — CHANGES_REQUESTED** [review]

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go (hunk 2 above)
Comment (mitaliy) [reply feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go]: I have made the changes.

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go (hunk 3 above)
Comment (mitaliy) [reply feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go]: I have removed the extra line.

**Review by Ankur19 — DISMISSED** [review]
LGTM! 

I have a doubt regarding the Replace. Deferring this to Quattro.

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go [hunk 5]
```diff
@@ -92,10 +92,15 @@ func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
...
-// configurePortId configures p4rt port-id on the DUT.
+// configurePortId configures p4rt port-id and interface type on the DUT.
 func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
 	portName := dut.Port(t, "port1").Name()
-	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portId)
+	currIntf := &oc.Interface{Name: ygot.String(portName),
+		Type: oc.IETFInterfaces_InterfaceType_ethernetCsmacd,
+		Id:   &portId}
+	gnmi.Replace(t, dut, d.Interface(portName).Config(), currIntf)
```
[the remark this answers is not in the document]
Comment (mitaliy) [reply feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go:102]: Hi Ankur, 

Yes it did run successfully.

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go [hunk 6]
```diff
@@ -92,10 +92,15 @@ func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
 	gnmi.Replace(t, dut, gnmi.OC().Component(p4rtNode).Config(), &c)
 }
 
-// configurePortId configures p4rt port-id on the DUT.
+// configurePortId configures p4rt port-id and interface type on the DUT.
 func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
 	portName := dut.Port(t, "port1").Name()
-	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portId)
+	currIntf := &oc.Interface{Name: ygot.String(portName),
```
Comment (greg-dennis) [inline feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go:99]: Let's make sure the braces and field indentation match (https://google.github.io/styleguide/go/decisions#matching-braces)

	currIntf := &oc.Interface{
		Name: ygot.String(portName),
		Type: oc.IETFInterfaces_InterfaceType_ethernetCsmacd,
		Id:   &portId,
	}

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go (hunk 6 above)
Comment (mitaliy) [reply feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go]: Hi Greg,
The issue has been addressed in the next commit. Thanks.

**Review by greg-dennis — APPROVED** [review]
approved for Go quality only

**Review by Ankur19 — APPROVED** [review]

**Review by sachendras — APPROVED** [review]

**Pull request merged**